lib/Runtime/Library/ForInObjectEnumerator.cpp

```diff
@@ -182,7 +207,32 @@ namespace Js
                 if (TestAndSetEnumerated(propertyId) //checks if the property is already enumerated or not
                     && (attributes & PropertyEnumerable))
                 {
-                    return currentIndex;
+                    bool propertyShadowed = false;
```

Comment:
Oh... I think you misunderstood.  I think should keep the optimization where the whole prototype chain has no enumerable properties. 

I am suggesting is that we don't skip any prototype if there are any enumerable property on the prototype chain, which I think is what you are suggesting, right?